collector/analyzer/pgftmetricanalyzer/pgft_analyzer.go

```diff
@@ -0,0 +1,137 @@
+	"github.com/Kindling-project/kindling/collector/model"
+	"github.com/Kindling-project/kindling/collector/model/constlabels"
+	"github.com/Kindling-project/kindling/collector/model/constnames"
+	"github.com/hashicorp/go-multierror"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+const (
+	PgftMetric analyzer.Type = "pgftmetricanalyzer"
+)
+
+var consumableEvents = map[string]bool{
+	constnames.SwitchEvent: true,
+}
+
+type PgftMetricAnalyzer struct {
+	consumers   []consumer.Consumer
+	conntracker conntrackerpackge.Conntracker
+	telemetry   *component.TelemetryTools
+}
+
+func NewPgftMetricAnalyzer(cfg interface{}, telemetry *component.TelemetryTools, nextConsumers []consumer.Consumer) analyzer.Analyzer {
+	retAnalyzer := &PgftMetricAnalyzer{
+		consumers: nextConsumers,
+		telemetry: telemetry,
+	}
+	conntracker, err := conntrackerpackge.NewConntracker(nil)
+	if err != nil {
+		telemetry.Logger.Warn("Conntracker cannot work as expected:", zap.Error(err))
+	}
+	retAnalyzer.conntracker = conntracker
+	return retAnalyzer
+}
+
+func (a *PgftMetricAnalyzer) Start() error {
+	return nil
+}
+
+// ConsumeEvent gets the event from the previous component
+func (a *PgftMetricAnalyzer) ConsumeEvent(event *model.KindlingEvent) error {
+	_, ok := consumableEvents[event.Name]
+	if !ok {
+		return nil
+	}
+	var gaugeGroup *model.GaugeGroup
+	var err error
+	switch event.Name {
+	case constnames.SwitchEvent:
+		gaugeGroup, err = a.generateSwitchPgft(event)
+	}
+	if err != nil {
+		if ce := a.telemetry.Logger.Check(zapcore.DebugLevel, "Event Skip, "); ce != nil {
+			ce.Write(
+				zap.Error(err),
+			)
+		}
+		return nil
+	}
+	if gaugeGroup == nil {
+		return nil
+	}
+	var retError error
+	for _, nextConsumer := range a.consumers {
+		err := nextConsumer.Consume(gaugeGroup)
+		if err != nil {
+			retError = multierror.Append(retError, err)
+		}
+	}
+	return retError
+}
+
+func (a *PgftMetricAnalyzer) generateSwitchPgft(event *model.KindlingEvent) (*model.GaugeGroup, error) {
+	labels, err := a.getSwitchLabels(event)
+	if err != nil {
+		return nil, err
+	}
+
+	pgftMaj := event.GetUserAttribute("pgft_maj")
+	pgftMin := event.GetUserAttribute("pgft_min")
+	ptMaj := (int64)(pgftMaj.GetUintValue())
+	ptMin := (int64)(pgftMin.GetUintValue())
+
+	var gaugeSlice []*model.Gauge
+	gaugeMaj := &model.Gauge{
+		Name:  constnames.PgftSwitchMajorMetricName,
+		Value: ptMaj,
+	}
+
+	gaugeMin := &model.Gauge{
+		Name:  constnames.PgftSwitchMinorMetricName,
+		Value: ptMin,
+	}
+	if ptMaj != 0 {
+		gaugeSlice = append(gaugeSlice, gaugeMaj)
+	}
+	if ptMin != 0 {
+		gaugeSlice = append(gaugeSlice, gaugeMin)
+	}
+
+	return model.NewGaugeGroup(constnames.PgftGaugeGroupName, labels, event.Timestamp, gaugeSlice...), nil
+}
+
+func (a *PgftMetricAnalyzer) getSwitchLabels(event *model.KindlingEvent) (*model.AttributeMap, error) {
+
+	ctx := event.GetCtx()
+	threadinfo := ctx.GetThreadInfo()
+	tid := (int64)(threadinfo.GetTid())
+	pid := (int64)(threadinfo.GetPid())
```

Comment:
Is it possible that `ctx` or `threadinfo` is nil? If yes, this will panic the program.